doc: Add missing top-level description to pruneblockchain RPC

nervana21 opened the pull request:
Previously, the `pruneblockchain` RPC help output included only the method signature and arguments, with no top-level description explaining its purpose or constraints.

This PR adds a top-level description, improving documentation consistency and alerting users to the potential impacts of using the command.

laanwj commented:
LGTM
```
pruneblockchain height

Deletes block and undo data up to a specified height or timestamp.
Requires `-prune` to be enabled at startup; this action is irreversible.

Arguments:
1. height    (numeric, required) The block height to prune up to. May be set to a discrete height, or to a UNIX epoch time
             to prune blocks whose block time is at least 2 hours older than the provided timestamp.

Result:
n    (numeric) Height of the last block pruned

Examples:
> bitcoin-cli pruneblockchain 1000
```

nervana21 committed:
doc: Add missing top-level description to pruneblockchain RPC

Previously, the `pruneblockchain` RPC help output included only the method signature and arguments, with no top-level description explaining its purpose or constraints.

This PR adds a concise top-level description, improving documentation consistency and alerting users to the potential impacts of using the command.

janb84 reviewed (commented):
ACK [fa5ed18](https://github.com/bitcoin/bitcoin/pull/32333/commits/fa5ed18946858c837c65b4fa730fbeda780c68e2)

This PR adds some additional explanatory text (top level) to the `pruneblockchain` RPC help output. Provides more clarity what the command does and makes it more consistent with other help outputs. 

New output : 
```shell
 % ./bitcoin-cli --regtest pruneblockchain
error code: -1
error message:
pruneblockchain height

Deletes block and undo data up to a specified height or timestamp.
Requires `-prune` to be enabled at startup; this action is irreversible.

Arguments:
1. height    (numeric, required) The block height to prune up to. May be set to a discrete height, or to a UNIX epoch time
             to prune blocks whose block time is at least 2 hours older than the provided timestamp.

Result:
n    (numeric) Height of the last block pruned

Examples:
> bitcoin-cli pruneblockchain 1000
```

luke-jr commented on src/rpc/blockchain.cpp:
```diff
@@ -854,7 +854,9 @@ std::optional<int> GetPruneHeight(const BlockManager& blockman, const CChain& ch
 {
-    return RPCHelpMan{"pruneblockchain", "",
+    return RPCHelpMan{"pruneblockchain",
+                "\nDeletes block and undo data up to a specified height or timestamp.\n"
```
...only if certain criteria are met?

luke-jr commented on src/rpc/blockchain.cpp:
```diff
@@ -854,7 +854,9 @@ std::optional<int> GetPruneHeight(const BlockManager& blockman, const CChain& ch
-    return RPCHelpMan{"pruneblockchain", "",
+    return RPCHelpMan{"pruneblockchain",
+                "\nDeletes block and undo data up to a specified height or timestamp.\n"
+                "Requires `-prune` to be enabled at startup; this action is irreversible.\n",
```
`getblockfrompeer` can reverse it AFAIK

nervana21 commented:
Thanks for the feedback :)

I’ve updated the description to clarify that pruning is conditional (not guaranteed), and that pruned data may be re-fetched in some cases (e.g., via getblockfrompeer). As such, the action is not strictly irreversible. Let me know if you'd like to see any further changes.

nervana21 commented on src/rpc/blockchain.cpp:
```diff
@@ -854,7 +854,9 @@ std::optional<int> GetPruneHeight(const BlockManager& blockman, const CChain& ch
 {
-    return RPCHelpMan{"pruneblockchain", "",
+    return RPCHelpMan{"pruneblockchain",
+                "\nAttempts to delete block and undo data up to a specified height or timestamp, if eligible for pruning.\n"
```
Updated, thanks!

janb84 reviewed (commented):
re ACK [135a0f0](https://github.com/bitcoin/bitcoin/commit/135a0f0aa711b95c50aa4cbe0c38d82d647f1c8b)

Changes since last ACK: 
- Textual improvements of description to be more precise.
- Removal of leading new line (/n)

yancyribbens commented:
cr ACK https://github.com/bitcoin/bitcoin/pull/32333/commits/135a0f0aa711b95c50aa4cbe0c38d82d647f1c8b

Nice, didn't know this rpc existed.

achow101 commented:
ACK 135a0f0aa711b95c50aa4cbe0c38d82d647f1c8b
